junit-jupiter-api/src/main/java/org/junit/jupiter/api/DisplayNameGenerator.java

```diff
@@ -141,4 +141,100 @@ private static boolean hasParameters(Method method) {
 		}
 	}
 
+	/**
+	 * {@code DisplayNameGenerator} that supports {@code ReplaceUnderscores} to generate complete sentences.
+	 *
+	 * <p>This generator extends the functionality of {@link ReplaceUnderscores} by
+	 * generating a human-readable display names that form complete sentences divided each
+	 * class, nested class and test by a ({@code ','}).
+	 *
+	 * @since 5.6
+	 */
+	@API(status = EXPERIMENTAL, since = "5.6")
+	class IndicativeSentencesGenerator extends ReplaceUnderscores {
+
+		@Override
+		public String generateDisplayNameForClass(Class<?> testClass) {
+			return removeRootCharacter(super.generateDisplayNameForClass(testClass));
+		}
+
+		@Override
+		public String generateDisplayNameForNestedClass(Class<?> nestedClass) {
+			return classReplaceToIndicativeSentence(nestedClass);
+		}
+
+		@Override
+		public String generateDisplayNameForMethod(Class<?> testClass, Method testMethod) {
+			return classReplaceToIndicativeSentence(testClass) + getSentenceSeparator(testClass)
+					+ super.generateDisplayNameForMethod(testClass, testMethod);
+		}
+
+		private String classReplaceToIndicativeSentence(Class<?> testClass) {
+			Class<?> classWithEnclosingParent = testClass.getEnclosingClass();
+			DisplayName classWithDisplayName = testClass.getAnnotation(DisplayName.class);
+			DisplayNameGeneration classWithAnnotation = testClass.getAnnotation(DisplayNameGeneration.class);
```

Comment:
Please use `AnnotationSupport.findAnnotation` in the above two lines to support meta annotations.

Reply:
Roger, applying changes to the following commit.

Reply:
Plus, the method returns an `Optional` so you might be able to remove some if statements below ;)

Reply:
Also, I think it would be more readable if you'd rename these variables `displayName` and `displayNameGeneration`